pkg/validation/policy/validate.go

```diff
@@ -471,8 +471,8 @@ func ruleForbiddenSectionsHaveVariables(rule *kyvernov1.Rule) error {
 
 // hasVariables - check for variables in the policy
 func hasVariables(policy kyvernov1.PolicyInterface) [][]string {
-	policy = cleanup(policy)
-	policyRaw, _ := json.Marshal(policy)
+	polCopy := cleanup(policy.CreateDeepCopy())
```

Comment:
deep copying the policy fixes the issue ?

Reply:
The error was "fatal error: concurrent map writes", the panic occurs when we reset one entry in the annotations map, so I created a copy of each policy so that it operates on the copied version.

Reply:
i'm not sure i follow what map is written concurrently ?

Reply:
>	github.com/kyverno/kyverno/pkg/validation/policy/validate.go:483 +0x50

The annotations map [here](https://github.com/kyverno/kyverno/blob/main/pkg/validation/policy/validate.go#L483).

Reply:
And we have concurrent go routines in the validation process ?

Reply:
Did you manage to reproduce the issue ?

Reply:
No, I did not try it locally :(